packages/angular_devkit/build_angular/src/webpack/plugins/scripts-webpack-plugin.ts

```diff
@@ -35,30 +35,33 @@ function addDependencies(compilation: any, scripts: string[]): void {
     compilation.fileDependencies.add(script);
   }
 }
-
-function hook(compiler: any, action: (compilation: any, callback: (err?: Error) => void) => void) {
-  compiler.hooks.thisCompilation.tap('scripts-webpack-plugin', (compilation: any) => {
-    compilation.hooks.additionalAssets.tapAsync(
-      'scripts-webpack-plugin',
-      (callback: (err?: Error) => void) => action(compilation, callback),
-    );
-  });
-}
-
 export class ScriptsWebpackPlugin {
   private _lastBuildTime?: number;
   private _cachedOutput?: ScriptOutput;
 
   constructor(private options: Partial<ScriptsWebpackPluginOptions> = {}) { }
 
-  shouldSkip(compilation: any, scripts: string[]): boolean {
+  async shouldSkip(compilation: any, scripts: string[]): Promise<boolean> {
     if (this._lastBuildTime == undefined) {
       this._lastBuildTime = Date.now();
       return false;
     }
 
-    for (let i = 0; i < scripts.length; i++) {
-      const scriptTime = compilation.fileTimestamps.get(scripts[i]);
+    for (const script of scripts) {
+      const scriptTime = isWebpackFiveOrHigher()
+        ? await new Promise<number | undefined>((resolve, reject) => {
+          compilation.fileSystemInfo.getFileTimestamp(script, (error: unknown, entry: any) => {
+            if (error) {
+              reject(error);
+
+              return;
+            }
+
+            resolve(typeof entry !== 'string' ? entry.safeTime : undefined)
```

Comment:
`entry` can be a string?

Reply:
Apparently, https://github.com/webpack/webpack/blob/302d079a75f8a18c3e4aee891c7751404dc6f239/types.d.ts#L3338-L3341